packages/kit/src/runtime/client/client.js

```diff
@@ -1363,3 +1354,27 @@ export function create_client({ target, base, trailing_slash }) {
 		}
 	};
 }
+
+let data_id = 1;
+
+/**
+ * @param {URL} url
+ * @param {boolean[]} invalid
+ * @returns {Promise<import('types').ServerData>}
+ */
+async function load_data(url, invalid) {
+	const data_url = new URL(url);
+	data_url.pathname += `${url.pathname.endsWith('/') ? '' : '/'}__data.js`;
+	data_url.searchParams.set('__invalid', invalid.map((x) => (x ? 'y' : 'n')).join(''));
+	data_url.searchParams.set('__id', String(data_id++));
+
+	await import(/* @vite-ignore */ data_url.href);
```

Comment:
could this be more sensitive to race conditions? For example you go to page A, then B right away, B is loaded faster, so you end up with A here? It might be safer to write to `window.__sveltekit_data_<id>` if that's possible or check if another request is pending.
In general I think a small comment why we do `await import` and `window._something` here might be good for future readers.

Reply:
It's not vulnerable to race conditions, because we read and delete the data in the same tick that the promise resolves. Something would have to happen in between the `await import(...)` and the `const server_data = ...` for it to be racy.

We can't use a unique ID for the data because that would fail in cases where the data was prerendered.

I'll add some comments

Reply:
ae423fd9abba9f6c04f0e718405b6f82b00f03aa